### PR TITLE
Add command for tagging commit

### LINKED ISSUE #103: tag commit
In the commit log, I want to be able to tag the selected commit.

* [ ] new command
* [ ] text input popup

### PR DESCRIPTION
**Pull request by cruessler**
Opened as a draft for getting early feedback.

Open questions:

- [x] Do you want to validate the tag before calling `repo.tag` or do you want
  to rely on git refusing invalid tags?
- [x] Should the dialog have just a single line?

Will close #103.

### CODE REVIEW

**Comment by extrawurst** [comment]
@cruessler thanks for tackling this. 

> Do you want to validate the tag before calling repo.tag or do you want
to rely on git refusing invalid tags?
I would rely on git refusing an invalid tag. we have a reusable error message box for that already.

> Should the dialog have just a single line?

Not exactly sure what you mean? I would expect the same dialog we already use for the stash-name.

File: asyncgit/src/sync/commit.rs [hunk 1]
```diff
@@ -80,6 +80,23 @@ pub fn commit(repo_path: &str, msg: &str) -> Result<CommitId> {
         .into())
 }
 
+pub fn tag(
```
Comment (extrawurst) [inline asyncgit/src/sync/commit.rs:83]: Please add a doc comment here and also a unites

File: src/tabs/revlog.rs [hunk 2]
```diff
@@ -28,6 +28,7 @@ const SLICE_SIZE: usize = 1200;
 ///
 pub struct Revlog {
     commit_details: CommitDetailsComponent,
+    tag_input: TextInputComponent,
```
Comment (extrawurst) [inline src/tabs/revlog.rs:31]: directly integrating this into the revlog is not the right approach. please check the `StashMsgComponent` that is register in `App` as a Popup.

**Comment by cruessler** [comment]
@extrawurst Thanks for the feedback! I have incorporated it and am now able to locally tag commits. The revlog does not update, though, after a commit is tagged and I have found no way to change that. `InternalEvent::Update(NeedsUpdate::ALL)` is pushed to the queue after the commit is tagged. Am I missing something?

**Comment by extrawurst** [comment]
@cruessler I could help you more when you would push those changes you made :)

**Comment by extrawurst** [comment]
@cruessler as of #195 new tags will be refetched every 2s.

when you add a tag you can also *force* a refetch (because we know a new one is gonna be there) in `Revlog`:
```
self.git_tags.request(Duration::from_secs(0), true)?;
```

**Comment by cruessler** [comment]
@extrawurst I updated the branch with your feedback. I did not need to explicitly ask the `Revlog` to re-load the tags, the showed up immediately.

**Comment by extrawurst** [comment]
@cruessler nice work! thank you! ❤️

**Pull request merged**